File: NeetCode - arrays and hashing/Challenge4.cpp
#include<bits/stdc++.h>

using namespace std;

class Solution {
public:
    vector<int> static twoSum(vector<int>& nums, int target) {
        map<int, int> number;
        int size = nums.size();

        for(int i = 0; i < size; ++i){
            int remainder = target - nums[i];
            if(number.count(remainder)){
                return vector<int>{number[remainder], i};
            }
            number[nums[i]] = i;
        }

        return vector<int>{-1, -1};
    }
};

int main(){
    vector<int> a = {-3,4,3,90};
    vector<int> sol = Solution::twoSum(a, 0);
    cout<<sol[0]<<" "<<sol[1];



    return 0;
}

File: NeetCode - Greedy/Challenge4.py
class Solution:
    def jump(self, nums: list[int]) -> int:
        count = 0
        curr = 0
        while curr < len(nums) - 1:
            count += 1
            indexOfMax = curr + 1

            jump = nums[curr]
            if curr + jump >= len(nums) - 1:
                break

            j = curr + 1
            while j <= curr + jump and j < len(nums):
                if indexOfMax + nums[indexOfMax] <= j + nums[j]:
                    indexOfMax = j

                j += 1
            curr = indexOfMax
        
        return count

sol = Solution()
nums = [2,4,1,1,1,1]
print(sol.jump(nums))